lib/ts/framework/utils.ts

```diff
@@ -23,7 +22,72 @@ import type { HTTPMethod } from "../types";
 import { COOKIE_HEADER } from "./constants";
 import { getFromObjectCaseInsensitive } from "../utils";
 import contentType from "content-type";
-import inflate from "inflation";
+import pako from "pako";
+import { decompress as brotliDecompress } from "brotli";
+
+async function inflate(stream: IncomingMessage): Promise<string> {
+    if (!stream) {
+        throw new TypeError("argument stream is required");
+    }
+
+    const encoding = (stream.headers && stream.headers["content-encoding"]) || "identity";
+
+    let decompressedData: Uint8Array | string;
+
+    if (encoding === "gzip" || encoding === "deflate") {
+        const inflator = new pako.Inflate();
+
+        for await (const chunk of stream) {
+            inflator.push(chunk, false);
+        }
+
+        if (inflator.err) {
+            throw new Error(`Decompression error: ${inflator.msg}`);
+        }
+
+        decompressedData = inflator.result;
+    } else if (encoding === "br") {
+        const chunks: Buffer[] = [];
+        for await (const chunk of stream) {
+            chunks.push(chunk);
+        }
+        const compressedData = Buffer.concat(chunks);
+        decompressedData = brotliDecompress(compressedData);
+    } else {
+        // Handle identity or unsupported encoding
+        decompressedData = Buffer.concat([]);
+        for await (const chunk of stream) {
+            decompressedData = Buffer.concat([decompressedData, chunk]);
+        }
+    }
+
+    if (typeof decompressedData === "string") return decompressedData;
+
+    return new TextDecoder().decode(decompressedData);
+}
+
+export function parseParams(string: string): object {
+    // Set up a new URLSearchParams object using the string.
+    const params = new URLSearchParams(string);
+
+    // Get an iterator for the URLSearchParams object.
+    const entries = params.entries();
+
+    const result: { [key: string]: any } = {};
+
+    // Loop through the URLSearchParams object and add each key/value
```

Comment:
Was this always done in the library we used to use?

Reply:
Just checked the source code, they do a `forEach` loop but yeah, this is done it seems.

Reply:
Are you sure about that? Could you provide a link?

Reply:
Updated it as per our discussion